app/controllers/application_controller.rb

```diff
@@ -5,6 +5,130 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "password_security"
   end
 
-end
\ No newline at end of file
+  get '/' do
+    erb :index
+  end
+
+  get '/tweets' do
+    @user = current_user if logged_in?
+    logged_in? ? (erb :'tweets/tweets') : (redirect to '/login')
+  end
+
+  get '/tweets/new' do
+    logged_in? ? (erb :'tweets/create_tweet') : (redirect to '/login')
+  end
+
+  post '/tweets' do
+    if logged_in? && (current_user.id == session[:id])
+      if params[:content].strip.empty?
+        redirect to '/tweets/new'
+      else
+        params[:user_id] = current_user.id
+        @tweet = Tweet.create(params)
+        redirect to '/tweets'
+      end
+    else
+      redirect to '/'
+    end
+  end
+
+  get '/tweets/:id' do
+    if logged_in?
+      @tweet = Tweet.find_by_id(params[:id])
+      !!@tweet ? (erb :'tweets/show_tweet') : (redirect to '/tweets')
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    @tweet = Tweet.find_by_id(params[:id])
+    if logged_in?
+      if current_user.id == @tweet.user_id
+        !!@tweet ? (erb :'tweets/edit_tweet') : (redirect to '/tweets')
+      else
+        redirect to '/tweets'
+      end
+    else
+      redirect to '/login'
+    end
+  end
+
+  patch '/tweets/:id' do
+    if params[:content].strip.empty?
+      redirect to "/tweets/#{params[:id]}/edit"
+    else
```

Comment:
I would contend that you still want to double check the user's association to the tweet here. There are other ways to submit a PATCH/POST request than going through an HTML form...